src/plugins/dashboard/public/dashboard_container/component/grid/dashboard_grid.tsx

```diff
@@ -6,123 +6,30 @@
  * Side Public License, v 1.
  */
 
-import _ from 'lodash';
-import sizeMe from 'react-sizeme';
-import classNames from 'classnames';
 import 'react-resizable/css/styles.css';
 import 'react-grid-layout/css/styles.css';
-import React, { useCallback, useMemo, useRef } from 'react';
-import ReactGridLayout, { Layout, ReactGridLayoutProps } from 'react-grid-layout';
 
-import { ViewMode, EmbeddablePhaseEvent } from '@kbn/embeddable-plugin/public';
+import { pick } from 'lodash';
+import classNames from 'classnames';
+import { useEffectOnce } from 'react-use/lib';
+import React, { useState, useMemo, useCallback } from 'react';
+import { Layout, Responsive as ResponsiveReactGridLayout } from 'react-grid-layout';
```

Comment:
The old `ReactGridLayout` that we were using was **not** truly responsive - its width could be changed, but we had to manually override this layouting for the "one column" mobile layout via CSS. However, the old`ReactGridLayout` has a **static** height that was generated a single time - this meant that, because we were not adjusting this height when switching to the single column layout, the height of the `ReactGridLayout` containers were not updated correctly.

So, when switching into mobile view with the sticky header, it would scroll **past the end** of the container, which would make the sticky header drift off screen like so: